endpoint.go

```diff
@@ -61,6 +61,7 @@ type endpoint struct {
 	generic       map[string]interface{}
 	joinLeaveDone chan struct{}
 	prefAddress   net.IP
+	prefAddressV6 net.IP
```

Comment:
I thought of that, but I felt it is better to have two separate fields for v4 and v6 otherwise we have to go through the list each time to find out whether an ipv6 was specified.

Also I feel it is better to keep the primary ip separated from the list of secondary: Actually in the code @jc-m shared with us, he is also keeping the secondary IPs in a separate list.
I agree with that, also because we may not want to enforce the <ip belongs to the subnet> for secondary ip and also discard the possible <already allocated error> to cover the use case where the secondary IP is a VIP shared by more than one container.

In summary, the allocation of the secondary IPs would likely take a different path than the allocation of the primary, therefore it is not ideal to keep all of them in the same list, even more to only have one list for v4 and v6.